### ISSUE TITLE
Github Actions status not displayed

### ISSUE THREAD

**Issue by lwasyl** [ptsochantaris/trailer#407]
I'm not sure if I'm missing some configuration piece, but for PRs in Trailer I only see status for outside jobs for which we update status using Github API. For Github Actions, despite having checks in status list, I can't see them in Trailer. Is this intended?

**Comment by ptsochantaris** [comment]
Hi @lwasyl and sorry for the late reply. Trailer doesn't yet support GitHub actions as it's a beta API and liable to change. As soon as GitHub makes it official I'll be looking to add them in ASAP. If you spot anything related to this please do give a shout!

**Comment by lwasyl** [comment]
Hey, no worries. Can you point me to where the beta API that would be useful for this feature is described? I only found some APIs that don't seem relevant.

**Comment by ptsochantaris** [comment]
This is the "hub" for the API: https://developer.github.com/v3/actions/

If you go into the various sections, you'll notice that endpoints are still marked as "beta" :( Hopefully this will change sooner rather than later!

**Comment by lwasyl** [comment]
I might be missing something, but the only `beta` notices in workflow-related APIs are next to the `usage` endpoints. I'm not sure I see an edpoint useful for fetching statuses for the given PR, but other than that the APIs seem to be stable

**Comment by carlisia** [comment]
Hi @ptsochantaris and @lwasyl 👋 

tl;dr: GH Actions API is out of beta: [GitHub Actions: API is now generally available | GitHub Changelog](https://github.blog/changelog/2020-03-24-github-actions-api-is-now-generally-available/) as of March 2020, so it should be solid!

If you click on the link posted here [Github Actions status not displayed · Issue #407 · ptsochantaris/trailer](https://github.com/ptsochantaris/trailer/issues/407), and search for "beta", you'll find only one end point is in beta, and that is to approve a workflow run, so doesn't apply.


-------
Hi @ptsochantaris, first of all, thank you! This tool saves my bacon on an hourly basis. I get paid to work on open source and have to track multiple repos. I was failing at that until I started using and learned to setup all the knobs.

I pay for apps that help me way less. Is there a way I could show my appreciation? I don't know Swift so contributing code is not in the cards right. now. Please let me know. I have always thought there should be a paid version of this app, and I'm taking the opportunity with this first post to say so! You may email me at carlisia@grokkingtech.io if you have any idea for a contribution I can make now.

Now, the reason I landed here: took me forever to realize the CI status was not showing, despite the fact I set it to show. It does show the status of Netlify, which is one of the CI checks, but none of the others. First I thought it is because Netlify is the last check, and Trailer was only picking up that last one. I set trailer to filter out the "netlify" string and then nothing showed.  Then I saw this issue, I started suspecting it is because the other checks run on GH Actions. 

This glitch had been leading me to think PRs were fine and ready for review, when really some were failing and needed a way more immediate action. This situation is now forcing me to have to go the the GH website anyway on noez.

**Comment by ptsochantaris** [comment]
Hi @carlisia and thank you for your very kind words about Trailer, they are much appreciated, and I'm really glad it's useful for you!

I certainly haven't given Trailer the love it needs for a few months now and part of that is keeping up with the GitHub Actions API evolution.

The info you have linked to above is very useful, and means I can get back to integrating the API now that it's stable (I was badly burned in the past by trying to keep up with beta APIs in GH) -- So, stay tuned, and many thanks for the nudge, I probably would not have noticed that :)

**Comment by carlisia** [comment]
Thank you!

**Comment by ptsochantaris** [comment]
Here's a build with run-check functionality enabled for v4 API. That API is quite straightforward and, now that it's stable, quite easy to add in.

Can you give it a go and let me know if what it displays makes sense? I'm not too familiar with actions and run checks, so it could be that Trailer is giving too much info, or not enough, or even not working at all :D 

There is no specific setting needed to turn support on for this, it's part of the general "show statuses" option.

[Trailer.app.zip](https://github.com/ptsochantaris/trailer/files/6744012/Trailer.app.zip)

**Comment by carlisia** [comment]
Glad to test it!

This first screen shot is of my original installation:
![2021-06-30 at 2 16 PM](https://user-images.githubusercontent.com/16508/124034500-24cfc600-d9b0-11eb-8b11-febe7bd67b97.png)

This second one is from the new installation. I waited for the refresh to finish. The second PR which has a green netlify status changed to yellow than back to green. The top one which shows as yellow never changed. In real life, that CI status didn't change, it is green.
![2021-06-30 at 2 33 PM](https://user-images.githubusercontent.com/16508/124034535-2f8a5b00-d9b0-11eb-861d-5fd52848c1a4.png)

**Comment by ptsochantaris** [comment]
Brilliant, thanks for testing! Seems like the update logic needs some additional tweaking. I’ll try testing using the repos above too 👍

**Comment by ptsochantaris** [comment]
Here's an update with some minor fixes which may or may not address the above issue - however from what I see above there are no checks displayed there, so possibly Trailer didn't refresh those PRs yet? I'm including a screenshot of what this repo should look like with the actions synced:

[Trailer.app.zip](https://github.com/ptsochantaris/trailer/files/6747250/Trailer.app.zip)

<img width="524" alt="Screenshot 2021-07-01 at 10 56 19" src="https://user-images.githubusercontent.com/521581/124105274-f955e580-da5a-11eb-9824-fae59d7ddd3f.png">

**Comment by carlisia** [comment]
Wow nice!!!! Oh boy, I'm glad you have the option to show only the failing ones, so many checks! I loooove that it shows the diff. And also that it shows there's a conflict, that's another super helpful indicator!

I installed it and am eagerly waiting for it to finish refreshing. Looks the same as yesterday so far except that the yellow check that should be green is now back to being green.

Is there a way to force a specific PR to refresh?

**Comment by carlisia** [comment]
After it finished refreshing it looks the same, none of the new indicators are showing. If it's only going to show with new PR's that's totally fine. I'll find out soon enough, will report.

**Comment by ptsochantaris** [comment]
Are you sure you are using the v4 API for GitHub? If not, the new changes won't show up - you can turn it on in preferences. The options to show the line diffs and conflict are also in the "display" tab if you want to turn them on 👍

**Comment by lwasyl** [comment]
I think there's some other unrelated issue with refreshing using v4 api. I use it:
<img width="558" alt="image" src="https://user-images.githubusercontent.com/3951580/124121488-3b8c2080-da75-11eb-87b9-4e23707df3f5.png">

and routinely see build statuses not refreshing. Also I have couple of repos which show PRs that have long been merged, and I can't seem to get rid of them. I believe (I'm like 70% sure) it started after checking the following option:
<img width="330" alt="image" src="https://user-images.githubusercontent.com/3951580/124121641-6d9d8280-da75-11eb-8559-84426f5d0288.png">

**Comment by carlisia** [comment]
Boooyah!!!!!!!

Yep. I did not have the v4 checked 🤣 

![2021-07-01 at 5 09 AM](https://user-images.githubusercontent.com/16508/124122179-a82dfd80-da2a-11eb-9495-5d04dfef7c8d.png)

![2021-07-01 at 5 10 AM](https://user-images.githubusercontent.com/16508/124122260-bf6ceb00-da2a-11eb-9626-7ca67f6295d9.png)

**Comment by carlisia** [comment]
I think the big ask of this issue which was to enable the GH API v4 functionalities in Tailer was addressed and this issue deserves to be closed, at least as far as I'm concerned. Any other related request probably would be best tracked in a new issue.

Thank you so so much @ptsochantaris for the quick turn around and for the app!

**Comment by ptsochantaris** [comment]
Awesome! No need to close the issue for now, I'll do it when the upcoming release is out 👍 

BTW here's another build with minor fixes to avoid red "X"s when querying statuses with actions, if there are a lot of PRs the previous build could break: 

[Trailer.app.zip](https://github.com/ptsochantaris/trailer/files/6749601/Trailer.app.zip)

**Comment by carlisia** [comment]
I installed this one but the X is still there.

![2021-07-01 at 4 37 PM](https://user-images.githubusercontent.com/16508/124200626-ac8a0300-da8a-11eb-8d09-e12a8fb44ac4.png)

**Comment by ptsochantaris** [comment]
Sorry, I wasn’t very clear there - the X indicators I was talking about were the ones that appear if a sync fails. The ones in the actions should indicate failed run-checks (but please do let me know in case they should not be there!)

**Comment by ptsochantaris** [comment]
Hi, I'll be closing this issue as v1.7.6 is now up and includes this amongst other changes. If you encounter any problems with it, please feel free to open a new issue.
